library/core/src/num/nonzero.rs

```diff
@@ -364,6 +367,23 @@ where
     #[rustc_const_stable(feature = "nonzero", since = "1.28.0")]
     #[must_use]
     #[inline]
+    #[rustc_allow_const_fn_unstable(const_refs_to_cell)]
```

Comment:
Is this necessary?

Reply:
Without it we are receiving compile errors which is due to the byte comparison we are executing in requires and ensures. We used this method because we were unable to compare the generic type.

Error:
```rust
error: const-stable function cannot use `#[feature(const_refs_to_cell)]`
   --> /volume/rustyrelic/verify-rust-std/library/core/src/num/nonzero.rs:370:5
    |
370 | /     #[requires({
371 | |         let size = core::mem::size_of::<T>();
372 | |         let ptr = &n as *const T as *const u8;
373 | |         let slice = unsafe { core::slice::from_raw_parts(ptr, size) };
374 | |         !slice.iter().all(|&byte| byte == 0)
375 | |     })]
    | |_______^
    |
    = note: this error originates in the attribute macro `kani_core::requires` (in Nightly builds, run with -Z macro-backtrace for more info)
help: if the function is not (yet) meant to be stable, make this function unstably const
    |
370 +     #[rustc_const_unstable(feature = "...", issue = "...")]
371 |     #[requires({
    |
help: otherwise, as a last resort `#[rustc_allow_const_fn_unstable]` can be used to bypass stability checks (but requires team approval)
    |
370 +     #[rustc_allow_const_fn_unstable(const_refs_to_cell)]
371 |     #[requires({
    |

error: aborting due to 1 previous error
```

Reply:
I see. Can you add a comment that mentions it's needed for the byte comparison?

Reply:
Added comments